File: ring-android/app/src/main/java/cx/ring/utils/ClipboardHelper.java
/*
 *  Copyright (C) 2004-2021 Savoir-faire Linux Inc.
 *
 *  Author: Romain Bertozzi <romain.bertozzi@savoirfairelinux.com>
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
package cx.ring.utils;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.text.TextUtils;
import android.util.Log;

import androidx.annotation.NonNull;

import cx.ring.BuildConfig;
import cx.ring.R;

public class ClipboardHelper {
    public static final String TAG = ClipboardHelper.class.getSimpleName();

    public static void copyToClipboard(final @NonNull Context context,
                                       final String text) {
        if (TextUtils.isEmpty(text)) {
            Log.d(TAG, "copyNumberToClipboard: number is null");
            return;
        }

        ClipboardManager clipboard = (ClipboardManager) context
                .getSystemService(Context.CLIPBOARD_SERVICE);
        ClipData clip = android.content.ClipData.newPlainText(context.getText(R.string.clip_contact_uri), text);
        clipboard.setPrimaryClip(clip);
    }
}

File: ring-android/app/src/main/java/cx/ring/utils/ClipboardHelper.kt
/*
 *  Copyright (C) 2004-2021 Savoir-faire Linux Inc.
 *
 *  Author: Romain Bertozzi <romain.bertozzi@savoirfairelinux.com>
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
package cx.ring.utils

import android.text.TextUtils
import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import cx.ring.R

object ClipboardHelper {
    val TAG = ClipboardHelper::class.simpleName!!
    fun copyToClipboard(
        context: Context,
        text: String?
    ) {
        if (TextUtils.isEmpty(text)) {
            return
        }
        val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        val clip = ClipData.newPlainText(context.getText(R.string.clip_contact_uri), text)
        clipboard.setPrimaryClip(clip)
    }
}